File: src/francaislang/ast/expressions/ConstValueExpr.java
package francaislang.ast.expressions;

import org.ascore.ast.buildingBlocs.Expression;
import org.ascore.lang.objects.ASCObject;

/**
 * Squelette de l'impl\u00E9mentation d'une expression.<br>
 * Pour trouver un exemple d'une impl\u00E9mentation compl\u00E8te
 *
 * @author Mathis Laroche
 */
public record ConstValueExpr(ASCObject<?> value) implements Expression<ASCObject<?>> {

    /**
     * Appel\u00E9 durant le Runtime, cette m\u00E9thode retourne un objet de type ASObjet
     *
     * @return le r\u00E9sultat de l'expression
     */
    @Override
    public ASCObject<?> eval() {
        return value;
    }
}

File: src/codemdr/ast/expressions/ConstValueExpr.kt
package codemdr.ast.expressions

import org.ascore.ast.buildingBlocs.Expression
import org.ascore.lang.objects.ASCObject


data class ConstValueExpr(private val value: ASCObject<*>) : Expression<ASCObject<*>> {
    /**
     * Appel\u00E9 durant le Runtime, cette m\u00E9thode retourne un objet de type ASObjet
     *
     * @return le r\u00E9sultat de l'expression
     */
    override fun eval(): ASCObject<*> {
        return value
    }
}
